ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3360,14 +3257,13 @@ public static Directory getAcidStateFromCache(Supplier<FileSystem> fileSystem,
 
     // compute and add to cache
     if (recompute || (value == null)) {
-      Directory dirInfo = getAcidState(fileSystem.get(), candidateDirectory, conf,
+      AcidDirectory dirInfo = getAcidState(fileSystem.get(), candidateDirectory, conf,
           writeIdList, useFileIds, ignoreEmptyFiles);
       value = new DirInfoValue(writeIdList.writeToString(), dirInfo);
 
       if (value.dirInfo != null && value.dirInfo.getBaseDirectory() != null
           && value.dirInfo.getCurrentDirectories().isEmpty()) {
         if (dirCacheDuration > 0) {
-          populateBaseFiles(dirInfo, useFileIds, fileSystem);
```

Comment:
Why was this here?

Reply:
The basefiles were populated and stored in cache, so they could be later used in OrcInputFormat if no write happens to the table before the next query. This is not necessary now, because ParsedBase will handle this when getFiles is called, if the baseFiles are not there it will do the listing once and store it. What is even better, if we are on S3 the files will be there already from the first recursive listing, so the FS is not even called once.